.github/workflows/deploy.yaml

```
@@ -10,20 +10,75 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
-    permissions:
-      pull-requests: write
-    env:
-      GH_TOKEN: ${{ github.token }}
     strategy:
       fail-fast: false
       matrix:
-        test: ["", check-external-links]
+        test: [""]
     steps:
       - name: Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
       - name: pre-commit checks
+        shell: bash
+        run: |
+          set -o pipefail
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends webp
+          pip install pre-commit
+
+          CI_FAIL=false
+          EXTRA_ARGS=("--all-files")
+
+          if [ "${{ github.event_name }}" == "push" ]
+          then
+            if [ "${{ github.event.created }}" != "true" ]
+            then
+              git fetch origin ${{ github.event.before }}:old_head
+              FROM_REF="${{ github.event.before }}"
+              TO_REF="${{ github.event.after }}"
+              EXTRA_ARGS=("--from-ref" "$FROM_REF" "--to-ref" "$TO_REF")
+            fi
+          elif [ "${{ github.event_name }}" == "pull_request" ]
+          then
+            FROM_REF="${{ github.event.pull_request.base.sha }}"
+            TO_REF="${{ github.event.pull_request.head.sha }}"
+            EXTRA_ARGS=("--from-ref" "$FROM_REF" "--to-ref" "$TO_REF")
+          elif [ "${{ github.event_name }}" == "merge_group" ]
+          then
+            git fetch origin ${{ github.event.merge_group.base_sha }}:old_head
+            FROM_REF="${{ github.event.merge_group.base_sha }}"
+            TO_REF="${{ github.event.merge_group.head_sha }}"
+            EXTRA_ARGS=("--from-ref" "$FROM_REF" "--to-ref" "$TO_REF")
+          fi
+
```

```suggestion

          if [[ ! -z "${{ matrix.test }}" ]]
          then
            EXTRA_ARGS+=("--hook-stage" "manual" "${{ matrix.test }}")
          fi

```
复制就是字面意义上的复制啊，咋还少东西了